src/components/input/baseInput/baseInput.tsx

```diff
@@ -138,86 +122,107 @@ export default function TextInput(props: TextInputProps) {
     if (e.key === 'Enter' && !e.shiftKey) {
       e.preventDefault()
-      handleSendMessage()
+      !isSendDisabled && handleSendMessage()
     }
   }
 
   function handleOnChange(e: React.ChangeEvent<HTMLInputElement>): void {
-    setErrorMessage('')
+    setSpeechToTextError('')
     setMessageText(e.target.value)
   }
 
   return (
-    <Box className="rustic-text-input-container">
-      <Box className="rustic-text-input-and-error-container">
-        {errorMessage.length > 0 && (
+    <Box className="rustic-base-input" ref={ref} data-cy="base-input">
+      <Box className="rustic-error-and-input-container">
+        <Box className="rustic-error-container">
           <Typography
             variant="caption"
             color="error"
             className="rustic-error-message"
+            data-cy="error-message"
           >
-            {errorMessage}
+            {speechToTextError}
           </Typography>
-        )}
-        <TextField
-          data-cy="text-input"
-          className="rustic-text-input"
-          variant="outlined"
-          value={messageText}
-          label={props.label}
-          placeholder={props.placeholder}
-          maxRows={props.maxRows}
-          multiline={props.multiline}
-          fullWidth={props.fullWidth}
-          onKeyDown={handleKeyDown}
-          onChange={handleOnChange}
-          onFocus={() => setIsFocused(true)}
-          onBlur={() => setIsFocused(false)}
-          inputRef={inputRef}
-          color="secondary"
-          size="small"
-          error={!!errorMessage}
-          InputProps={
-            props.enableSpeechToText ? speechToTextButtonAdornment : {}
-          }
-        />
+        </Box>
+        <Box
+          className="rustic-input-field"
+          sx={{
+            border: '1px solid',
+            borderColor: isFocused ? 'secondary.main' : 'action.disabled',
+          }}
+        >
+          <TextField
+            data-cy="text-field"
+            className="rustic-text-field"
+            variant="outlined"
+            value={messageText}
+            label={props.label}
+            placeholder={props.placeholder}
+            maxRows={props.maxRows}
+            multiline={props.multiline}
+            fullWidth={props.fullWidth}
+            onKeyDown={handleKeyDown}
+            onChange={handleOnChange}
+            onFocus={() => setIsFocused(true)}
+            onBlur={() => setIsFocused(false)}
+            inputRef={inputRef}
+            color="secondary"
+            size="small"
+            InputProps={
+              props.enableSpeechToText ? speechToTextButtonAdornment : {}
+            }
+            InputLabelProps={{
+              className: !isFocused ? 'rustic-input-label' : '',
+              sx: {
+                backgroundColor: 'background.paper',
+              },
+            }}
+          />
+          <div className="rustic-file-preview-container"></div>
```

Comment:
This component has no idea about file previews. I think we should rename the class to something else - like `rustic-input-extras` or something. Since in future, this may be used by other features/functionality.